pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -173,11 +171,10 @@ func (nm *NodeManager) DiscoverNode(node *v1.Node) error {
 	return nil
 }
 
-
-
 func (nm *NodeManager) RegisterNode(node *v1.Node) error {
 	nm.addNode(node)
-	return nm.DiscoverNode(node)
+	nm.DiscoverNode(node)
```

Comment:
I think the earlier logic is correct, the caller can decide what to do with the error if discover node fails.

Reply:
Sandeep this is the original code. I just took out changes I did.
We don't need any change in node manager for this task.